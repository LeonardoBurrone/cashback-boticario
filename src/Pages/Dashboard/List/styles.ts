import styled from 'styled-components';

const Information = styled('div')`
  && {
    align-items: baseline;
    display: flex;
    justify-content: space-between;
  }
`;

export { Information };
