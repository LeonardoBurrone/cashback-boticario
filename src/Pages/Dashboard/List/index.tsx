import { Grid, Divider, Typography } from '@material-ui/core';
import moment from 'moment';
import React from 'react';

import { Information } from '../styles';

import { Purchase } from '../../../Ducks/Dashboard/types';
import { CBContent } from '../../../Styles/Common';

type Props = {
  purchases: Purchase[];
};

const List: React.FunctionComponent<Props> = (props: Props) => {
  const translateStatus = {
    Approved: 'Aprovado',
    Rejected: 'Rejeitado',
    Validating: 'Em validação'
  };

  const renderInformation = (description: string, value: string) => {
    return (
      <Information>
        <Typography variant="subtitle1">{description}</Typography>
        <Typography variant="subtitle2">{value}</Typography>
      </Information>
    );
  };

  return (
    <React.Fragment>
      {props.purchases.map((purchase: Purchase, index: number) => {
        return (
          <Grid key={`${purchase.id}${index}`} item xs={10}>
            <CBContent>
              {renderInformation('Código', `${purchase.id}`)}
              <Divider />
              {renderInformation('Valor da compra', `R$${purchase.price}`)}
              <Divider />
              {renderInformation('Data', `${moment(purchase.date).format('DD/MM/YYYY')}`)}
              <Divider />
              {renderInformation('% de cashback', `${purchase.cashbackPercentage}%`)}
              <Divider />
              {renderInformation('Valor do cashback', `R$${purchase.cashbackValue}`)}
              <Divider />
              {renderInformation('Status', `${translateStatus[purchase.status]}`)}
            </CBContent>
          </Grid>
        );
      })}
    </React.Fragment>
  );
};

export default List;
