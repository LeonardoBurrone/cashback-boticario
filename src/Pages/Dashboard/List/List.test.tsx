import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import List, { Props } from './';

describe('Test page List', () => {
  const props: Props = {
    purchases: [
      {
        cashbackPercentage: 0.01,
        cashbackValue: 1,
        date: '2021-04-01T08:15:30-03:00',
        id: 123,
        price: 100,
        status: 'Approved'
      },
      {
        cashbackPercentage: 0.01,
        cashbackValue: 1,
        date: '2021-04-01T08:11:30-03:00',
        id: 123,
        price: 50,
        status: 'Validating'
      },
      {
        cashbackPercentage: 0.01,
        cashbackValue: 1,
        date: '2021-04-01T08:19:30-03:00',
        id: 123,
        price: 30,
        status: 'Rejected'
      }
    ]
  };

  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<List {...props} />);
  });

  it('Check if component rendered correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
