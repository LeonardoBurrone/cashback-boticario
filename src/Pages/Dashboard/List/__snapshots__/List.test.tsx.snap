// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test page List Check if component rendered correctly 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Grid))
    item={true}
    key="1230"
    xs={10}
  >
    <Styled(WithStyles(ForwardRef(Paper)))>
      <styled.div>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle1"
        >
          Código
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle2"
        >
          123
        </WithStyles(ForwardRef(Typography))>
      </styled.div>
      <WithStyles(ForwardRef(Divider)) />
      <styled.div>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle1"
        >
          Valor da compra
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle2"
        >
          R$100
        </WithStyles(ForwardRef(Typography))>
      </styled.div>
      <WithStyles(ForwardRef(Divider)) />
      <styled.div>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle1"
        >
          Data
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle2"
        >
          01/04/2021
        </WithStyles(ForwardRef(Typography))>
      </styled.div>
      <WithStyles(ForwardRef(Divider)) />
      <styled.div>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle1"
        >
          % de cashback
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle2"
        >
          0.01%
        </WithStyles(ForwardRef(Typography))>
      </styled.div>
      <WithStyles(ForwardRef(Divider)) />
      <styled.div>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle1"
        >
          Valor do cashback
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle2"
        >
          R$1
        </WithStyles(ForwardRef(Typography))>
      </styled.div>
      <WithStyles(ForwardRef(Divider)) />
      <styled.div>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle1"
        >
          Status
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle2"
        >
          Aprovado
        </WithStyles(ForwardRef(Typography))>
      </styled.div>
    </Styled(WithStyles(ForwardRef(Paper)))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    key="1231"
    xs={10}
  >
    <Styled(WithStyles(ForwardRef(Paper)))>
      <styled.div>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle1"
        >
          Código
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle2"
        >
          123
        </WithStyles(ForwardRef(Typography))>
      </styled.div>
      <WithStyles(ForwardRef(Divider)) />
      <styled.div>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle1"
        >
          Valor da compra
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle2"
        >
          R$50
        </WithStyles(ForwardRef(Typography))>
      </styled.div>
      <WithStyles(ForwardRef(Divider)) />
      <styled.div>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle1"
        >
          Data
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle2"
        >
          01/04/2021
        </WithStyles(ForwardRef(Typography))>
      </styled.div>
      <WithStyles(ForwardRef(Divider)) />
      <styled.div>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle1"
        >
          % de cashback
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle2"
        >
          0.01%
        </WithStyles(ForwardRef(Typography))>
      </styled.div>
      <WithStyles(ForwardRef(Divider)) />
      <styled.div>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle1"
        >
          Valor do cashback
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle2"
        >
          R$1
        </WithStyles(ForwardRef(Typography))>
      </styled.div>
      <WithStyles(ForwardRef(Divider)) />
      <styled.div>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle1"
        >
          Status
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle2"
        >
          Em validação
        </WithStyles(ForwardRef(Typography))>
      </styled.div>
    </Styled(WithStyles(ForwardRef(Paper)))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    key="1232"
    xs={10}
  >
    <Styled(WithStyles(ForwardRef(Paper)))>
      <styled.div>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle1"
        >
          Código
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle2"
        >
          123
        </WithStyles(ForwardRef(Typography))>
      </styled.div>
      <WithStyles(ForwardRef(Divider)) />
      <styled.div>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle1"
        >
          Valor da compra
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle2"
        >
          R$30
        </WithStyles(ForwardRef(Typography))>
      </styled.div>
      <WithStyles(ForwardRef(Divider)) />
      <styled.div>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle1"
        >
          Data
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle2"
        >
          01/04/2021
        </WithStyles(ForwardRef(Typography))>
      </styled.div>
      <WithStyles(ForwardRef(Divider)) />
      <styled.div>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle1"
        >
          % de cashback
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle2"
        >
          0.01%
        </WithStyles(ForwardRef(Typography))>
      </styled.div>
      <WithStyles(ForwardRef(Divider)) />
      <styled.div>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle1"
        >
          Valor do cashback
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle2"
        >
          R$1
        </WithStyles(ForwardRef(Typography))>
      </styled.div>
      <WithStyles(ForwardRef(Divider)) />
      <styled.div>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle1"
        >
          Status
        </WithStyles(ForwardRef(Typography))>
        <WithStyles(ForwardRef(Typography))
          variant="subtitle2"
        >
          Rejeitado
        </WithStyles(ForwardRef(Typography))>
      </styled.div>
    </Styled(WithStyles(ForwardRef(Paper)))>
  </WithStyles(ForwardRef(Grid))>
</Fragment>
`;
